src/writer.rs

```diff
@@ -330,6 +333,48 @@ impl<'a, W: Write> ElementWriter<'a, W> {
             .write_event(Event::End(self.start_tag.to_end()))?;
         Ok(self.writer)
     }
+
+    /// Serialize an arbitrary value inside the current element
+    #[cfg(feature = "serialize")]
+    pub fn write_serializable_content<T: Serialize>(self, content: T) -> Result<&'a mut Writer<W>> {
```

Comment:
@Mingun I'm not sure which usage you are referring to, there is no `with_content()` method only `write_content()` and other methods following the same scheme.

https://github.com/tafia/quick-xml/blob/master/src/writer.rs#L285-L332

The naming scheme currently used here matches with that, whereas your suggestion does not.  

I'm not sure if using `DeResult` is a good idea because it would then be inconsistent with the other methods.

Reply:
I think I disagree slightly with the overall approach here, we should support this directly on the writer as opposed to on `ElementWriter`.  If you want to use it in conjunction with `ElementWriter` it would be easy with `write_inner_content()`

https://github.com/tafia/quick-xml/blob/master/src/writer.rs#L285-L332

Reply:
e.g.

```Rust
        writer
            .create_element("paired")
            .with_attribute(("attr1", "value1"))
            .with_attribute(("attr2", "value2"))
            .write_inner_content(|writer| {
                writer.write_serializable(&content)
            })?;
```

Reply:
> @Mingun I'm not sure which usage you are referring to, but the scheme you suggest is inconsistent with all of the existing methods such as `write_text_content()`, `write_empty()`, etc.

I mean, that we have `with_attribute`, so another `with_*` looks familiar. But you're right -- `with_*` methods does not end the chain. Ok, let it be `write_serializable_content`.

> Likewise using `DeResult` would be inconsistent with the other methods.

That looks not so inconsistent, if we took into account that this is another serialization level which have its own error type. I do not like hiding real errors in library code -- the caller should decide how to handle them. We cannot include `DeError` into `Error` because `DeError` already includes `Error`. So the logical step is just return `DeError` here.

> I think I disagree slightly with the overall approach here, we should support this directly on the writer as opposed to on `ElementWriter`.

I would see this as an auxiliary method for a frequent use case. We can have methods both on the `Writer` and `ElementWriter`.